src/database/databaseFactory.js

```diff
@@ -0,0 +1,28 @@
+import mongodb from './adapter/mongo.js';
+
+export default class DatabaseFactory {
```

Comment:
This class does not really match a factory pattern. This creates Connections, not Databases.
Accordingly, think `DatabaseConnectionFactory` would be more fitting in this case.